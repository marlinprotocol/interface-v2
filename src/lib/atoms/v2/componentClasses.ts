const baseFilledButtonClasses =
	'btn btn-theme btn-sm items-center flex justify-center disabled:cursor-not-allowed !rounded-[100px]';

export const buttonClasses = {
	filled: `${baseFilledButtonClasses} btn-primary disabled:bg-primary disabled:bg-opacity-60 disabled:text-white`,
	greyFilled: `${baseFilledButtonClasses} bg-base-200 text-primary border-none hover:bg-base-300`,
	lightblueFilled: `${baseFilledButtonClasses} btn-lightblue`,
	whiteFilled: `${baseFilledButtonClasses} bg-white text-black border border-[#D9DADE] hover:bg-base-200`,
	outlined:
		'btn btn-theme btn-sm text-primary px-4 border-primary bg-inherit hover:text-primary-focus hover:border-primary-focus hover:bg-inherit disabled:bg-inherit disabled:text-grey-300 disabled:border-grey-300 disabled:cursor-not-allowed',
	greyOutlined: `${baseFilledButtonClasses} border-[#D9DADE] bg-inherit hover:text-primary-focus hover:border-primary-focus hover:bg-inherit disabled:bg-inherit disabled:text-grey-300 disabled:border-grey-300 disabled:cursor-not-allowed`,
	error:
		'btn btn-theme btn-sm text-error px-5 border-error bg-inherit hover:bg-inherit hover:border-error hover:text-error-focus disabled:cursor-not-allowed',
	text: 'btn btn-sm btn-ghost p-0 bg-transparent hover:bg-transparent disabled:bg-transparent disabled:cursor-not-allowed',
	icon: 'btn btn-xs btn-ghost btn-circle p-0 bg-transparent hover:bg-transparent flex disabled:cursor-not-allowed',
	iconLightBlue:
		'btn btn-xs btn-circle p-0 bg-base-300 hover:bg-base-300 border-none disabled:bg-base-200 disabled:cursor-not-allowed',
	dropdownIcon:
		'btn btn-xs rounded-full btn-circle p-0 bg-base-300 border-primary/[.2] hover:bg-base-300 disabled:bg-base-200 disabled:cursor-not-allowed',
	maxButton:
		'btn btn-sm btn-text btn-ghost p-0 bg-transparent hover:bg-transparent font-bold text-primary tracking-widest disabled:text-primary disabled:cursor-not-allowed',
	largeButton:
		'flex gap-2 text-primary h-14 text-base font-semibold items-center disabled:cursor-not-allowed',
	tableConvertButton:
		'btn btn-xs bg-primary text-xs h-[35px] w-[126px] rounded btn-primary disabled:bg-primary disabled:bg-opacity-60 disabled:text-white disabled:cursor-not-allowed rounded-[100px] font-inter font-medium',
	paginationButton:
		'btn btn-sm px-2 border-none bg-base-200 hover:bg-base-300 disabled:bg-base-200 disabled:text-primary disabled:curson-not-allowed disabled:cursor-not-allowed'
};

export const tableClasses = {
	heading: 'text-primary text-sm font-semibold px-2 pb-4',
	row: 'border-b border-[#D9DADE] font-normal whitespace-nowrap last:border-b-0',
	cell: 'text-left text-sm leading-none',
	cellMini: 'text-xs py-1',
	empty: 'text-base text-center p-4 w-full'
};

export const inputClasses = {
	inputText:
		'input input-ghost h-[30px] w-full text-body p-0 font-normal text-[#030115] disabled:opacity-70 disabled:text-[#030115] disabled:placeholder:text-[#030115]/[.2] focus-within:text-[#030115] placeholder:text-[#030115]/[.2] focus:outline-none focus-within:border-b-2 focus:bg-transparent',
	searchInputText:
		'input input-ghost h-[30px] w-full text-base p-0 font-normal text-[#030115] focus:outline-none focus-within:border-b-2 focus-within:text-[#030115] focus:bg-transparent disabled:text-blue-500 placeholder-[#A8A8A8]'
};
