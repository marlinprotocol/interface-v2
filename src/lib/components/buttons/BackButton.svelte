<script lang="ts">
	import { staticImages } from '$lib/components/images/staticImages';

	export let text = 'Back';
	export let href: string;
</script>

<a data-testid="back-button" class="flex gap-3" {href}>
	<img src={staticImages.LeftBlueArrow} alt="Back" width="12" />
	{text}
</a>
