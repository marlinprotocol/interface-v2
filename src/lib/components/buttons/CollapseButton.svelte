<script lang="ts">
	import { staticImages } from '$lib/components/images/staticImages';
	import { cn } from '$lib/utils/helpers/commonHelper';

	export let disabled = false;
	export let onclick: (() => void) | undefined = undefined;
	export let isOpen = false;
	export let id = '';
</script>

<button data-testid="collapse-button" {id} on:click={onclick} {disabled} class="w-max">
	<img
		class={cn('rotate-0 transform transition duration-200', { 'rotate-180': isOpen })}
		src={staticImages.chevronDownIcon}
		alt="Collapse Icon"
	/>
</button>
