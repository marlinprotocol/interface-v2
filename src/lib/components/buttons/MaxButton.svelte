<script lang="ts">
	import { buttonClasses } from '$lib/atoms/componentClasses';
	import { cn } from '$lib/utils/helpers/commonHelper';

	export let disabled = false;
	export let onclick: (() => void) | undefined = undefined;
	export let styleClass: string = '';
</script>

<button on:click={onclick} {disabled} class={cn(buttonClasses.maxButton, styleClass)}> MAX </button>
