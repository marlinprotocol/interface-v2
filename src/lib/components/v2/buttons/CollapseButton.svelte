<script lang="ts">
	import { buttonClasses } from '$lib/atoms/componentClasses';
	import Icon from '$lib/atoms/icons/Icon.svelte';
	import { staticImages } from '$lib/components/images/staticImages';
	import chevronDown from 'svelte-awesome/icons/chevronDown';
	import chevronUp from 'svelte-awesome/icons/chevronUp';

	export let disabled = false;
	export let onclick: (() => void) | undefined = undefined;
	export let isOpen = false;
	export let styleClass = '';
	export let id = '';

	$: chevronIcon = isOpen ? chevronUp : chevronDown;
</script>

<button
	data-testid="collapse-button"
	{id}
	on:click={onclick}
	class="{styleClass} {buttonClasses.iconLightBlue}"
	{disabled}
>
	<!-- <img src={staticImages.chevronDownIcon}  /> -->
	<Icon data={chevronIcon} size={12} iconColorClass={disabled ? 'icon-disabled' : 'icon-primary'} />
</button>
