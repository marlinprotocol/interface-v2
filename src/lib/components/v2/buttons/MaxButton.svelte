<script lang="ts">
	import { buttonClasses } from '$lib/atoms/v2/componentClasses';

	export let disabled = false;
	export let onclick: (() => void) | undefined = undefined;
</script>

<button on:click={onclick} {disabled} class={buttonClasses.maxButton}> MAX </button>
