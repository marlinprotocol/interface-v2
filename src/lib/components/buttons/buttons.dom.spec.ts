import { describe, it, expect } from 'vitest';
import { render, fireEvent } from '@testing-library/svelte';
import BackButton from './BackButton.svelte';
import MaxButton from './MaxButton.svelte';
import CollapseButton from './CollapseButton.svelte';
import { buttonClasses } from '$lib/atoms/componentClasses';

describe('Back button', () => {
	it('renders the back button component properly', () => {
		const { container } = render(BackButton);
		expect(container).toMatchSnapshot();
	});

	it('renders correctly with initial props', () => {
		const { getByText, getByRole } = render(BackButton, {
			props: {
				href: '#',
				text: 'Back'
			}
		});
		expect(getByText('Back')).toBeTruthy();
		const anchor = getByRole('link');
		expect(anchor.getAttribute('href')).toBe('#');
	});

	it('renders the correct image', () => {
		const { getByAltText } = render(BackButton, {
			props: {
				href: '#',
				text: 'Back'
			}
		});

		const image = getByAltText('Back');
		const staticImagesImported = { LeftBlueArrow: '/images/left-blue-arrow.svg' };
		expect(image.getAttribute('src')).toBe(staticImagesImported.LeftBlueArrow);
	});

	it('renders with correct classes', () => {
		const { getByTestId } = render(BackButton, {
			props: {
				href: '#',
				text: 'Back'
			}
		});

		const anchorButton = getByTestId('back-button');
		expect(anchorButton.className).toBe('flex gap-3');
	});
});

describe('CollapseButton', () => {
	it('renders the collapse button component properly', () => {
		const { container } = render(CollapseButton);
		expect(container).toMatchSnapshot();
	});

	it('renders correctly with initial props', async () => {
		const mockFn = vi.fn(() => {});
		const testingId = 'testing-collapse-button';
		const { getByTestId } = render(CollapseButton, {
			props: {
				disabled: false,
				onclick: mockFn,
				id: testingId
			}
		});
		const button = getByTestId('collapse-button') as HTMLButtonElement;
		expect(button).toBeTruthy();
		expect(button.className).contain('w-max');
		expect(button.id).contain(testingId);
		expect(button.disabled).toBe(false);
	});

	it('fires an onClick event if disabled false', async () => {
		const mockFn = vi.fn(() => {});
		const testingId = 'testing-collapse-button';
		const { getByTestId } = render(CollapseButton, {
			props: {
				disabled: false,
				onclick: mockFn,
				id: testingId
			}
		});

		// Simulate a click event on the button.
		const button = getByTestId('collapse-button');
		await fireEvent.click(button);

		// Check if the mock function was called.
		expect(mockFn).toHaveBeenCalledTimes(1);
	});

	it("onClick doesn't work if disabled true", async () => {
		const mockFn = vi.fn(() => {});
		const testingId = 'testing-collapse-button';
		const { getByTestId } = render(CollapseButton, {
			props: {
				disabled: true,
				onclick: mockFn,
				id: testingId
			}
		});

		// Simulate a click event on the button.
		const button = getByTestId('collapse-button');
		button.click();

		// Check if the mock function was called.
		expect(mockFn).toHaveBeenCalledTimes(0);
	});
});

describe('Max Button', () => {
	it('renders the max button component properly', () => {
		const { container } = render(MaxButton);
		expect(container).toMatchSnapshot();
	});

	it('renders with correct text and class', () => {
		const { getByText } = render(MaxButton);

		const button = getByText('MAX');
		expect(button).toBeTruthy(); // Button renders with text 'MAX'
		expect(button.className).toBe(buttonClasses.maxButton); // Button has correct class
	});

	it('is disabled when the disabled prop is true', () => {
		const { getByText } = render(MaxButton, { props: { disabled: true } });

		const button = getByText('MAX') as HTMLButtonElement;
		expect(button.disabled).toBe(true); // Button is disabled
	});

	it('calls the onclick function when clicked', async () => {
		const mockFn = vi.fn();
		const { getByText } = render(MaxButton, { props: { onclick: mockFn } });

		const button = getByText('MAX');
		button.click();

		expect(mockFn).toHaveBeenCalledTimes(1); // onclick mock function is called once
	});

	it('does not call onclick function when disabled', async () => {
		const mockFn = vi.fn();
		const { getByText } = render(MaxButton, { props: { disabled: true, onclick: mockFn } });

		const button = getByText('MAX');
		button.click();

		expect(mockFn).toHaveBeenCalledTimes(0); // onclick mock function is not called
	});
});
