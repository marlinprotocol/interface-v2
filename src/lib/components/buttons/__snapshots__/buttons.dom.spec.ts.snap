// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Back button > renders the back button component properly 1`] = `
<body>
  <div>
    <a
      class="flex gap-3"
      data-testid="back-button"
    >
      <img
        alt="Back"
        src="/images/left-blue-arrow.svg"
        width="12"
      />
       
      Back
    </a>
    <!--&lt;BackButton&gt;-->
  </div>
</body>
`;

exports[`CollapseButton > renders the collapse button component properly 1`] = `
<body>
  <div>
    <button
      class="w-max"
      data-testid="collapse-button"
      id=""
    >
      <img
        alt="Collapse Icon"
        class="rotate-0 transform transition duration-200"
        src="/images/chevronDownIcon.svg"
      />
    </button>
    <!--&lt;CollapseButton&gt;-->
  </div>
</body>
`;

exports[`Max Button > renders the max button component properly 1`] = `
<body>
  <div>
    <button
      class="btn btn-sm btn-text btn-ghost p-0 bg-transparent hover:bg-transparent font-bold text-primary tracking-widest disabled:text-primary disabled:cursor-not-allowed"
    >
      MAX
    </button>
    <!--&lt;MaxButton&gt;-->
  </div>
</body>
`;
